### DISCUSSION TITLE
Filename too long error for windows-2019 checkout

### DISCUSSION THREAD

**Discussion by omindu** [community/community#26952] [Actions]
<p>When using “actions/checkout@v1” on windows-2019. The job is failing with the error ’ <strong>cannot create directory at ### Filename too long</strong>’.</p>
<pre><code>steps:
    - uses: actions/checkout@v1
      with:
        ref: master
</code></pre>
<p>Is there a way to enable longpaths support for the job on windows?</p>

**Comment by omindu** [comment, marked as answer]
<p>Managed to achieve this by using the below</p>
<pre><code>steps:
    - name: Support longpaths
      run: git config --system core.longpaths true
</code></pre>

**Reply by HannesWell** [reply]
Exactly that command didn't work for me, because the system configuration isn't accessible (anymore?) in a GH-workflow.
But adjusting the global configuration instead, as suggested in https://github.com/actions/checkout/issues/242#issuecomment-627479980, did work in a workflow:
`git config --global core.longpaths true`

**Comment by alansingfield** [comment]
<p>You can use this to just apply longpaths when you do the cloning.</p>
<p>There are reasons why longpaths isn’t enabled globally.</p>
<p>git clone -c core.longpaths=true <a href="https://yourrepo.git" rel="nofollow noopener">https://yourrepo.git</a></p>
